### PR TITLE
CS-1150 Lambda Function for Sugar Portal Chat

### PR DESCRIPTION
**Pull request by jwelborn-sugar**
This PR adds an API Gateway, and Lambda Function to serve as an endpoint to start chat client from portal chat.

### CODE REVIEW

File: __tests__/unit/utils/aws/s3-utils.test.js [hunk 1]
```diff
@@ -49,16 +49,16 @@ describe('Tests getObjectKeyFromS3Event', function() {
     it('should get the object key from S3 trigger event', async () => {
         let testKey = 'test_key';
         let event = {
-            "Records": [
+            'Records': [
```
Comment (jwelborn-sugar) [inline __tests__/unit/utils/aws/s3-utils.test.js:52]: eslint changes

File: src/utils/aws/chat-utils.js [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+    if (body.hasOwnProperty('InstanceId')) {
+        instanceId = body['InstanceId'];
+    }
+    console.log('Instance ID: ' + instanceId);
+
+    return new Promise(function(resolve, reject) {
+        let startChat = {
+            'InstanceId': instanceId == '' ? process.env.instanceId : instanceId,
+            'ContactFlowId': contactFlowId == '' ? process.env.contactFlowId : contactFlowId,
+            'Attributes': body['Attributes'],
```
Comment (jwelborn-sugar) [inline src/utils/aws/chat-utils.js:35]: This is the only line changed from the provided amazon sample function. The provided sample was hard-coded to only set `customerName` in the attributes, but we'll be able to set multiple attributes with this.

**Review by sugarhqi — DISMISSED** [review]

File: template.yml [hunk 3]
```diff
@@ -30,6 +30,16 @@ Parameters:
     Type: String
     Default: ''
     Description: AWS Connect instance name
+  ContactFlowId:
+    Type: String
+    Default: 12345678-1234-1234-1234-123456789012
```
Comment (rogeryang-sugarcrm) [inline template.yml:35]: I think we should remove this default value. If the user does not enter a value, we should not try to access this contact flow

File: template.yml [hunk 4]
```diff
@@ -30,6 +30,16 @@ Parameters:
...
+    Description: The contact flow id that the customer will interact with while chatting
+    AllowedPattern: '\w{8}-\w{4}-\w{4}-\w{4}-\w{12}'
+  InstanceId:
+    Type: String
+    Default: 12345678-1234-1234-1234-123456789012
```
Comment (rogeryang-sugarcrm) [inline template.yml:40]: I think we should remove the default value for similar reasons above

**Review by rogeryang-sugarcrm — CHANGES_REQUESTED** [review]

**Review by rogeryang-sugarcrm — DISMISSED** [review]
Looks good to me

**Review by rogeryang-sugarcrm — APPROVED** [review]

**Pull request merged**